cider/common/Arena.h

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ * Copyright (c) 2016-2022 ClickHouse, Inc.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#pragma once
+
+#include <cstring>
+#include <memory>
+#include <vector>
+
+#include <boost/noncopyable.hpp>
+
+namespace DB {
+
+/** Memory pool to append something. For example, short strings.
+ * Usage scenario:
+ * - put lot of strings inside pool, keep their addresses;
+ * - addresses remain valid during lifetime of pool;
+ * - at destruction of pool, all memory is freed;
+ * - memory is allocated and freed by large MemoryChunks;
+ * - freeing parts of data is not possible (but look at ArenaWithFreeLists if you need);
+ */
+class Arena : private boost::noncopyable {
```

Comment:
do we need an ArenaAllocator like https://github.com/duckdb/duckdb/blob/master/src/include/duckdb/storage/arena_allocator.hpp

Reply:
Something like that but here it will only be used for string. Added some todos here for further implementation.